app/js/lib/foxbox/recipes.js

```diff
@@ -6,10 +6,35 @@
 const p = Object.freeze({
   // Private properties.
   settings: Symbol('settings'),
-  net: Symbol('net')
+  net: Symbol('net'),
+
+  // Recipe private properties.
+  service: Symbol('service')
 });
 
-export default class Recipe {
+export class Recipe {
+  constructor(service) {
+    if (!service) {
+      throw new Error('Service is required!');
+    }
+
+    this[p.service] = service;
+  }
+
+  get id() {
+    return this[p.service].id;
+  }
+
+  get label() {
+    return this[p.service].source && this[p.service].source.name;
```

Comment:
As it turned out we can have recipe without source (if something went wrong when we were creating a recipe). Likely we'll get rid of it later, but right now it saves a lot of development time - while experimenting with the taxonomy requests there is a high chance to create broken recipe :)